gcs/src/main/scala/monix/connect/gcs/configuration/BucketConfig.scala

```diff
@@ -0,0 +1,98 @@
+package monix.connect.gcs.configuration
+
+import com.google.cloud.storage.BucketInfo.{IamConfiguration, LifecycleRule}
+import com.google.cloud.storage.{Acl, BucketInfo, Cors, StorageClass}
+import io.monix.connect.gcs.configuration.BucketConfig.Location
+
+import scala.concurrent.duration.FiniteDuration
+import scala.jdk.CollectionConverters._
+
+final case class BucketConfig(name: String,
+                              location: Option[Location] = None,
+                              labels: Map[String, String] = Map.empty[String, String],
+                              requesterPays: Option[Boolean] = None,
+                              versioningEnabled: Option[Boolean] = None,
+                              storageClass: Option[StorageClass] = None,
+                              retentionPeriod: Option[FiniteDuration] = None,
+                              acl: List[Acl] = List.empty[Acl],
+                              cors: List[Cors] = List.empty[Cors],
+                              defaultAcl: List[Acl] = List.empty[Acl],
+                              lifecycleRules: List[LifecycleRule] = List.empty[LifecycleRule],
+                              logging: Option[BucketInfo.Logging] = None,
+                              indexPage: Option[String] = None,
+                              notFoundPage: Option[String] = None,
+                              defaultKmsKeyName: Option[String] = None,
+                              defaultEventBasedHold: Option[Boolean] = None,
+                              iamConfiguration: Option[IamConfiguration] = None) {
+
+  private[gcs] def getBucketInfo: BucketInfo = {
+    val builder = BucketInfo.newBuilder(name)
+    location.foreach(builder.setLocation)
+    storageClass.foreach(builder.setStorageClass)
+    logging.foreach(builder.setLogging)
+    retentionPeriod.foreach(rp => builder.setRetentionPeriod(rp.toMillis))
+    defaultEventBasedHold.foreach(evb => builder.setDefaultEventBasedHold(evb))
+
+    // Booleans
+    versioningEnabled.foreach(b => builder.setVersioningEnabled(b))
+    requesterPays.foreach(b => builder.setRequesterPays(b))
+
+    // Security and Access Control
+    builder.setAcl(acl.asJava)
+    builder.setDefaultAcl(defaultAcl.asJava)
+    builder.setCors(cors.asJava)
+    builder.setLifecycleRules(lifecycleRules.asJava)
+    iamConfiguration.foreach(builder.setIamConfiguration)
+    defaultKmsKeyName.foreach(builder.setDefaultKmsKeyName)
+
+    // Pages and Metadata
+    builder.setLabels(labels.asJava)
+    indexPage.foreach(builder.setNotFoundPage)
+    notFoundPage.foreach(builder.setNotFoundPage)
+
+    builder.build()
+  }
+}
+
+object BucketConfig {
+
+  type Location = String
+
+  object Locations {
+
+    // Regions
+    lazy val `northamerica-northeast-1`: Location = "NORTHAMERICA-NORTHEAST1"
```

Comment:
do we really need to define all these locations? there is no any google library where we could take them?

Reply:
I can have a look to see if there is a library that contains all of them. The JavaDoc points at their official list:

```java
/**
     * Sets the bucket's location. Data for blobs in the bucket resides in physical storage within
     * this region. A list of supported values is available <a
     * href="https://cloud.google.com/storage/docs/bucket-locations">here</a>.
     */
```

Reply:
https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/compute/Regions.html
here it is :)

Reply:
Thank you, can't believe I missed it :-/.

Reply:
@paualarco looking through the linked Enum it seems it's not up-to-date with the latest `Regions` and is also not inclusive of `Multi-Regions` and `Dual-Regions`, which seem specific to Google Cloud Storage.

I am also hesitant to add a dependency to the `Compute Engine` SDK just for an Enum. I would rather add something like Enumeratum and define the list of regions like that, or simply keep them as is.

Reply:
How can that not be up to date coming from google and a service that is used by so many people?  :O
It's very confusing :(